**XGB**

+----+--------------------------------------+----------+---------------+----------------+------------+--------------+-------------+----------+  
|    | predictor_func                       | origin   | destination   |   elapsed_time |   r2_score |   msle_score |   max_error |     mape |  
|----+--------------------------------------+----------+---------------+----------------+------------+--------------+-------------+----------|  
|  0 | <function predict at 0x7f4a60a674d0> | MOW      | LED           |       0.555244 |  -1.38772  |  0.00487748  |     332.926 |  6.19832 |  
|  1 | <function predict at 0x7f4a60a674d0> | MOW      | KZN           |       0.52695  |  -5.03453  |  0.000368194 |     160.021 |  1.64118 |  
|  2 | <function predict at 0x7f4a60a674d0> | MOW      | CEK           |       0.544231 |   0.73577  |  0.000884335 |     365.784 |  2.30079 |  
|  3 | <function predict at 0x7f4a60a674d0> | MOW      | SVX           |       0.546767 |   0.29796  |  0.00128515  |     368.996 |  2.06173 |  
|  4 | <function predict at 0x7f4a60a674d0> | MOW      | AER           |       0.572541 |   0.521279 |  0.00758033  |     622.832 |  8.3596  |  
|  5 | <function predict at 0x7f4a60a674d0> | MOW      | KRR           |       0.220395 |   0.268218 |  0.00290323  |     512.24  |  4.21521 |  
|  6 | <function predict at 0x7f4a60a674d0> | MOW      | KGD           |       0.21965  |  -0.882658 |  0.0145063   |    1560.88  |  7.49692 |  
|  7 | <function predict at 0x7f4a60a674d0> | LED      | MOW           |       0.535295 |   0        |  0.000659559 |     171.99  |  1.57109 |  
|  8 | <function predict at 0x7f4a60a674d0> | LED      | KZN           |       0.547624 |  -0.435788 |  0.00315502  |     585.215 |  5.16889 |  
|  9 | <function predict at 0x7f4a60a674d0> | LED      | CEK           |       0.558169 |  -1.76267  |  0.000998267 |     405.71  |  2.52739 |  
| 10 | <function predict at 0x7f4a60a674d0> | LED      | SVX           |       0.536686 |  -1.42738  |  0.0117553   |    1457.67  |  7.21987 |  
| 11 | <function predict at 0x7f4a60a674d0> | LED      | AER           |       0.548974 |  -0.236441 |  0.00653891  |    1023.95  |  7.04919 |  
| 12 | <function predict at 0x7f4a60a674d0> | LED      | KRR           |       0.224223 |  -1.34959  |  0.0372537   |    2030.47  | 13.1026  |  
| 13 | <function predict at 0x7f4a60a674d0> | LED      | KGD           |       0.224843 |  -1.39735  |  0.030225    |    1918.56  | 14.4467  |  
| 14 | <function predict at 0x7f4a60a674d0> | MOW      | SGC           |       0.228747 |  -2.53793  |  0.00217718  |     953.42  |  3.44607 |  
+----+--------------------------------------+----------+---------------+----------------+------------+--------------+-------------+----------+  

**Arima**

+----+-----------------------------------------------------+----------+---------------+----------------+-------------+--------------+-------------+----------+  
|    | predictor_func                                      | origin   | destination   |   elapsed_time |    r2_score |   msle_score |   max_error |     mape |  
|----+-----------------------------------------------------+----------+---------------+----------------+-------------+--------------+-------------+----------|  
|  0 | <function predict_queried_prices at 0x7f6912542290> | MOW      | LED           |        3.93446 |  -0.324811  |   0.00834292 |         709 |  7.03029 |  
|  1 | <function predict_queried_prices at 0x7f6912542290> | MOW      | KZN           |        6.06501 | -11.7459    |   0.00573717 |         851 |  6.21645 |  
|  2 | <function predict_queried_prices at 0x7f6912542290> | MOW      | CEK           |        3.14681 |  -2.58625   |   0.0191598  |        1176 | 12.796   |  
|  3 | <function predict_queried_prices at 0x7f6912542290> | MOW      | SVX           |        3.6844  | -17.6158    |   0.030769   |        1656 | 17.3064  |  
|  4 | <function predict_queried_prices at 0x7f6912542290> | MOW      | AER           |        3.17617 |  -1.24755   |   0.0516137  |        2399 | 19.2411  |  
|  5 | <function predict_queried_prices at 0x7f6912542290> | MOW      | KRR           |        3.88002 |  -7.75401   |   0.0276937  |        1844 | 15.5066  |  
|  6 | <function predict_queried_prices at 0x7f6912542290> | MOW      | KGD           |        4.74653 |  -1.0287    |   0.02448    |        1660 | 13.401   |  
|  7 | <function predict_queried_prices at 0x7f6912542290> | LED      | MOW           |        6.38617 | -15.6816    |   0.00631817 |         308 |  7.34463 |  
|  8 | <function predict_queried_prices at 0x7f6912542290> | LED      | KZN           |        4.49792 | -19.8816    |   0.0672398  |        2416 | 21.9465  |  
|  9 | <function predict_queried_prices at 0x7f6912542290> | LED      | CEK           |        4.23984 | -63.6088    |   0.324351   |        6476 | 48.6876  |  
| 10 | <function predict_queried_prices at 0x7f6912542290> | LED      | SVX           |        2.8912  |  -1.43199   |   0.023578   |        1791 |  9.94907 |  
| 11 | <function predict_queried_prices at 0x7f6912542290> | LED      | AER           |        2.87544 |  -0.457777  |   0.00880229 |        1235 |  8.26934 |  
| 12 | <function predict_queried_prices at 0x7f6912542290> | LED      | KRR           |        3.90492 |   0.0361094 |   0.0162326  |        1746 | 11.2964  |  
| 13 | <function predict_queried_prices at 0x7f6912542290> | LED      | KGD           |        4.91287 |  -5.08571   |   0.0736091  |        3022 | 20.1107  |  
| 14 | <function predict_queried_prices at 0x7f6912542290> | MOW      | SGC           |        3.42118 |  -3.29452   |   0.00583928 |        1653 |  5.11986 |  
+----+-----------------------------------------------------+----------+---------------+----------------+-------------+--------------+-------------+----------+  

In [23]:
from sklearn.metrics import r2_score, mean_squared_log_error, max_error

import pandas as pd
import numpy as np
import datetime
import prophet
from time import perf_counter

from sktime.forecasting.model_selection import temporal_train_test_split
from sktime.performance_metrics.forecasting import sMAPE, smape_loss


from tabulate import tabulate

import warnings
warnings.filterwarnings('ignore')

date_parser = lambda ts: datetime.strptime(ts, "%y-%m-%dT%H:%M:%SZ")
df = pd.read_csv('aviasales_data_t.csv')
df['requested_at'] = pd.to_datetime(df['requested_at'], format="%y-%m-%dT%H:%M:%SZ", errors="coerce")
df['departure_at'] = pd.to_datetime(df['departure_at'], format="%Y-%m-%dT%H:%M:%SZ", errors="coerce")
df['expires_at'] = pd.to_datetime(df['expires_at'], format="%Y-%m-%dT%H:%M:%SZ", errors="coerce")
df['price'] = pd.to_numeric(df["price"], errors="coerce")
df = df.dropna()
df = df.drop_duplicates(subset=['departure_at', 'expires_at', 'airline', 'flight_number', 'price'], keep='first')


def mean_absolute_percentage_error(y_true, y_pred): 
    y_true, y_pred = np.array(y_true), np.array(y_pred)
    return np.mean(np.abs((y_true - y_pred) / y_true)) * 100


# Profile prediction func and test prediction quality
def ml_model_test(orgn : str, dest : str,
                  current_date : datetime.datetime, flight_date : datetime.datetime,
                  predictor_f, target_f):
    print(orgn, dest)

    tic = perf_counter()
    predictions = predictor_f(orgn, dest, current_date, flight_date)
    tac = perf_counter()
    elapsed_time = tac - tic
    
    targets = target_f(orgn, dest, current_date, flight_date)
    print(f"Elapsed time: {elapsed_time:0.4f} seconds")
    print(predictions[1], targets[1])
    
    preds = list(map(lambda x: 1e-5 if x <= 0 else x, predictions[1]))
    r2 = r2_score(targets[1], preds)
    msle = mean_squared_log_error(targets[1], preds)
    max_err = max_error(targets[1], preds)
    mape = mean_absolute_percentage_error(targets[1], preds)
    
    print("R2:", r2)
    print("MSLE:", msle)
    print("Max error:", max_err)
    print("MAPE:", mape)
    
    return (str(predictor_f), orgn, dest, elapsed_time, r2, msle, max_err, mape)

# INSERT YOUR FUNCTION
def predict_queried_prices(orgn, dest, current_date, flight_date):
    days = []
    predictions = []
    delta = flight_date - current_date
    for i in range(delta.days + 1):
        days.append((flight_date - datetime.timedelta(days=i)).strftime("%Y-%m-%d"))
        predictions.append(predict_queried_prices_for(orgn, dest, current_date, flight_date, i))
    return days, predictions

def predict_queried_prices_for(orgn, dest, current_date, flight_date, n_days):
    
    orgn_dest = df[(df["origin"] == orgn) & (df["destination"] == dest) & (df["departure_at"] <= flight_date) 
                   & (df["requested_at"] >= df["departure_at"] - datetime.timedelta(days=n_days + 1))
                   & (df["requested_at"] <= df["departure_at"] - datetime.timedelta(days=n_days))] 
    
    orgn_dest_day_min = orgn_dest.resample('D', on='requested_at')['price'].min()
  
    orgn_dest_day_min = orgn_dest_day_min.fillna(orgn_dest_day_min.mean())  # not very smart
    
    y = orgn_dest_day_min
    y_train, y_test = temporal_train_test_split(y)
#     print(y_train, y_test, type(y_train))
#     return None

    prophet_train_df = pd.DataFrame(y_train)
    prophet_train_df.columns = ['y']
    prophet_train_df['ds'] = prophet_train_df.index
    
    index = pd.DataFrame(y_test)
    index.columns = ['y']
    index['ds'] = index.index
    del index['y']
    
    forecaster = prophet.Prophet()
    forecaster.fit(prophet_train_df)
#     fh = ForecastingHorizon(y_test.index, is_relative=False)
    y_pred = forecaster.predict(index)
#     print(y_pred)
    return round(y_pred.iloc[-1]['yhat'])

# INSERT YOUR FUNCTION
def show_real_prices(orgn, dest, current_date, flight_date, delta_days=7):
    
    orgn_dest = df[(df["origin"] == orgn) & (df["destination"] == dest) & (df["departure_at"] <= flight_date + datetime.timedelta(days=1))
                   & (df["departure_at"] >= flight_date)]
    
    days = []
    prices = []
    for i in range(delta_days + 1):
        orgn_dest_day = orgn_dest[(orgn_dest["requested_at"] >= orgn_dest["departure_at"] - datetime.timedelta(days=i + 1))
                           & (orgn_dest["requested_at"] <= orgn_dest["departure_at"] - datetime.timedelta(days=i))]
        if orgn_dest_day.empty:
            orgn_dest_day = orgn_dest[(orgn_dest["expires_at"] >= orgn_dest["departure_at"] - datetime.timedelta(days=i + 1))
                               & (orgn_dest["expires_at"] <= orgn_dest["departure_at"] - datetime.timedelta(days=i))]
        
        prices.append(round(orgn_dest_day["price"].min()))
        days.append((flight_date - datetime.timedelta(days=i)).strftime("%Y-%m-%d"))
        
    return days, prices



# # JUST AN EXAMPLE

i = "MOW"
ml_test_results = []
for i in ['MOW', 'LED']: # NaNs in others , 'KZN', 'CEK', 'SVX', 'AER', 'KRR', 'KGD'
    for j in ['MOW', 'LED', 'KZN', 'CEK', 'SVX', 'AER', 'KRR', 'KGD']: # , 'SGC'
        if i != j:
            ml_test_results.append(ml_model_test(i, j, datetime.datetime(2021,4,20), datetime.datetime(2021,4,27),
                                             predict_queried_prices, show_real_prices))
ml_test_results.append(ml_model_test("MOW", "SGC", datetime.datetime(2021,4,20), datetime.datetime(2021,4,27),
                                             predict_queried_prices, show_real_prices))            
            
ml_test_df = pd.DataFrame(ml_test_results)
ml_test_df.columns = ['predictor_func', 'origin', 'destination', 'elapsed_time', 'r2_score', 'msle_score', 'max_error', 'mape']
print(tabulate(ml_test_df, headers = 'keys', tablefmt = 'psql'))

    

INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.


MOW LED


INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to overr

Elapsed time: 6.8365 seconds
[4610, 2780, 2678, 2770, 2768, 2748, 3151, 3148] [3221, 2498, 2595, 2531, 2890, 2655, 2890, 3148]
R2: -3.0004823846435347
MSLE: 0.0199441987804052
Max error: 1389
MAPE: 10.476133925386003
MOW KZN


INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to overr

Elapsed time: 7.2054 seconds
[5362, 4679, 4206, 4103, 3917, 4089, 4440, 4525] [4498, 4172, 4248, 4226, 4172, 4172, 4248, 4248]
R2: -14.015751070823788
MSLE: 0.006912364207065858
Max error: 864
MAPE: 6.800294479267715
MOW CEK


INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to overr

Elapsed time: 6.9261 seconds
[6020, 4002, 3664, 5345, 5624, 4524, 5687, 6140] [5460, 4998, 4998, 5777, 6148, 6148, 5635, 5817]
R2: -3.4283689812962477
MSLE: 0.03328467047360277
Max error: 1624
MAPE: 13.220834126603249
MOW SVX


INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to overr

Elapsed time: 7.1046 seconds
[1717, 3838, 4093, 3740, 3667, 3746, 4248, 4394] [4058, 3658, 3953, 3658, 3658, 3658, 3658, 4058]
R2: -22.363402724610207
MSLE: 0.09655835446186926
Max error: 2341
MAPE: 11.931651207399273
MOW AER


INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to overr

Elapsed time: 7.4216 seconds
[4040, 6160, 5313, 5394, 4963, 5772, 6007, 5461] [3955, 4123, 5190, 5190, 5586, 5359, 6201, 6163]
R2: -0.10231391037577353
MSLE: 0.024842283741653585
Max error: 2037
MAPE: 11.404269610737991
MOW KRR


INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:n_changepoints greater than number of observations. Using 19.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:n_changepoints greater than number of observations. Using 19.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:n_changepoints greater than number of observations. Using 19.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_se

Elapsed time: 6.7025 seconds
[5033, 4935, 4876, 4953, 5588, 5622, 6060, 5791] [4998, 4998, 5320, 4998, 4869, 4365, 5363, 5363]
R2: -2.823093204659493
MSLE: 0.013960229628581835
Max error: 1257
MAPE: 9.46852720497597
MOW KGD


INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:n_changepoints greater than number of observations. Using 20.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:n_changepoints greater than number of observations. Using 19.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:n_changepoints greater than number of observations. Using 19.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_se

Elapsed time: 6.4762 seconds
[6613, 5721, 6178, 7149, 6365, 6708, 7682, 7783] [4820, 5920, 6428, 6428, 6428, 7000, 5618, 6876]
R2: -1.5209144866474715
MSLE: 0.028645809201118907
Max error: 2064
MAPE: 13.843477641972346
LED MOW


INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to overr

Elapsed time: 7.0305 seconds
[3699, 3224, 2755, 2677, 2841, 2896, 2956, 3106] [3058, 2890, 2890, 2890, 2890, 2890, 2890, 2890]
R2: -24.89407191448008
MSLE: 0.00778604397147929
Max error: 641
MAPE: 7.027616032370705
LED KZN


INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to overr

Elapsed time: 7.0768 seconds
[5778, 4803, 4792, 4590, 4396, 5020, 4996, 5298] [6823, 6845, 7070, 7582, 7724, 7362, 7362, 7888]
R2: -43.20330239909743
MSLE: 0.1661093790686158
Max error: 3328
MAPE: 32.087694317543495
LED CEK


INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to overr

Elapsed time: 6.8981 seconds
[-5607, 3400, 2614, 3362, 3549, 3388, 4294, 4286] [6736, 6668, 6629, 6880, 7257, 7996, 7060, 6510]
R2: -80.31002493841899
MSLE: 10.151555587281857
Max error: 6735.99999
MAPE: 55.34709106720202
LED SVX


INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to overr

Elapsed time: 6.8660 seconds
[6687, 5564, 5237, 5776, 5738, 5729, 6227, 6964] [7126, 6802, 7563, 6190, 6190, 7563, 6190, 7988]
R2: -2.2990363643003295
MSLE: 0.03573918257692478
Max error: 2326
MAPE: 13.346621249819485
LED AER


INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to overr

Elapsed time: 6.9750 seconds
[5457, 5508, 6479, 6677, 6639, 7071, 7609, 8174] [7982, 7188, 8397, 8375, 9098, 9026, 9098, 9098]
R2: -7.549652877233575
MSLE: 0.06702539676457016
Max error: 2525
MAPE: 21.66649518879242
LED KRR


INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:n_changepoints greater than number of observations. Using 20.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:n_changepoints greater than number of observations. Using 19.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:n_changepoints greater than number of observations. Using 19.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_se

Elapsed time: 6.3012 seconds
[4806, 6559, 7305, 7668, 7858, 7314, 7891, 7882] [6192, 6192, 6192, 6110, 7974, 8313, 8063, 7582]
R2: -0.0012520656819090537
MSLE: 0.020620008640643825
Max error: 1558
MAPE: 11.418337775672093
LED KGD


INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:n_changepoints greater than number of observations. Using 19.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:n_changepoints greater than number of observations. Using 19.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:n_changepoints greater than number of observations. Using 19.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_se

Elapsed time: 6.3014 seconds
[3020, 3772, 4034, 4686, 5630, 5826, 6765, 6427] [6808, 7265, 8735, 8753, 8443, 7316, 7316, 6803]
R2: -14.741522618422577
MSLE: 0.28777395152845253
Max error: 4701
MAPE: 33.84307173470135
MOW SGC


INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:n_changepoints greater than number of observations. Using 20.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:n_changepoints greater than number of observations. Using 19.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_seasonality=True to override this.
INFO:prophet:n_changepoints greater than number of observations. Using 19.
INFO:prophet:Disabling yearly seasonality. Run prophet with yearly_seasonality=True to override this.
INFO:prophet:Disabling daily seasonality. Run prophet with daily_se

Elapsed time: 6.4542 seconds
[9660, 8893, 9574, 8749, 8250, 8049, 7534, 7987] [7900, 8290, 8290, 8290, 7640, 7640, 7900, 8550]
R2: -6.861540952974152
MSLE: 0.01056434635421712
Max error: 1760
MAPE: 9.391629958868366
+----+-----------------------------------------------------+----------+---------------+----------------+--------------+--------------+-------------+----------+
|    | predictor_func                                      | origin   | destination   |   elapsed_time |     r2_score |   msle_score |   max_error |     mape |
|----+-----------------------------------------------------+----------+---------------+----------------+--------------+--------------+-------------+----------|
|  0 | <function predict_queried_prices at 0x7f26fadcca60> | MOW      | LED           |        6.83654 |  -3.00048    |   0.0199442  |        1389 | 10.4761  |
|  1 | <function predict_queried_prices at 0x7f26fadcca60> | MOW      | KZN           |        7.20535 | -14.0158     |   0.00691236 |         8

**Prophet**

+----+-----------------------------------------------------+----------+---------------+----------------+--------------+--------------+-------------+----------+  
|    | predictor_func                                      | origin   | destination   |   elapsed_time |     r2_score |   msle_score |   max_error |     mape |  
|----+-----------------------------------------------------+----------+---------------+----------------+--------------+--------------+-------------+----------|  
|  0 | <function predict_queried_prices at 0x7f26fadcca60> | MOW      | LED           |        6.83654 |  -3.00048    |   0.0199442  |        1389 | 10.4761  |  
|  1 | <function predict_queried_prices at 0x7f26fadcca60> | MOW      | KZN           |        7.20535 | -14.0158     |   0.00691236 |         864 |  6.80029 |  
|  2 | <function predict_queried_prices at 0x7f26fadcca60> | MOW      | CEK           |        6.92614 |  -3.42837    |   0.0332847  |        1624 | 13.2208  |  
|  3 | <function predict_queried_prices at 0x7f26fadcca60> | MOW      | SVX           |        7.10462 | -22.3634     |   0.0965584  |        2341 | 11.9317  |  
|  4 | <function predict_queried_prices at 0x7f26fadcca60> | MOW      | AER           |        7.42155 |  -0.102314   |   0.0248423  |        2037 | 11.4043  |  
|  5 | <function predict_queried_prices at 0x7f26fadcca60> | MOW      | KRR           |        6.70246 |  -2.82309    |   0.0139602  |        1257 |  9.46853 |  
|  6 | <function predict_queried_prices at 0x7f26fadcca60> | MOW      | KGD           |        6.47616 |  -1.52091    |   0.0286458  |        2064 | 13.8435  |  
|  7 | <function predict_queried_prices at 0x7f26fadcca60> | LED      | MOW           |        7.03049 | -24.8941     |   0.00778604 |         641 |  7.02762 |  
|  8 | <function predict_queried_prices at 0x7f26fadcca60> | LED      | KZN           |        7.07681 | -43.2033     |   0.166109   |        3328 | 32.0877  |  
|  9 | <function predict_queried_prices at 0x7f26fadcca60> | LED      | CEK           |        6.89806 | -80.31       |  10.1516     |        6736 | 55.3471  |  
| 10 | <function predict_queried_prices at 0x7f26fadcca60> | LED      | SVX           |        6.86599 |  -2.29904    |   0.0357392  |        2326 | 13.3466  |  
| 11 | <function predict_queried_prices at 0x7f26fadcca60> | LED      | AER           |        6.97504 |  -7.54965    |   0.0670254  |        2525 | 21.6665  |  
| 12 | <function predict_queried_prices at 0x7f26fadcca60> | LED      | KRR           |        6.30123 |  -0.00125207 |   0.02062    |        1558 | 11.4183  |  
| 13 | <function predict_queried_prices at 0x7f26fadcca60> | LED      | KGD           |        6.3014  | -14.7415     |   0.287774   |        4701 | 33.8431  |  
| 14 | <function predict_queried_prices at 0x7f26fadcca60> | MOW      | SGC           |        6.45417 |  -6.86154    |   0.0105643  |        1760 |  9.39163 |  
+----+-----------------------------------------------------+----------+---------------+----------------+--------------+--------------+-------------+----------+  

**Вывод:** Никакого смысла использовать prophet в этой задаче нет. Результаты даже хуже чем у AutoArima, а время работы сопоставимое.  
Лучшая модель -- на основе градиентного бустинга XGBoost.